### PR TITLE
Dcnm phase1 enhancements

### PR DESCRIPTION
**Pull request by kharicha**
status of vrf should be included when query vrf - https://github.com/CiscoDevNet/ansible-dcnm/issues/15
status of network should be included when query - https://github.com/CiscoDevNet/ansible-dcnm/issues/16

Fixed the above and the respective new IT cases have been added for both.
Fixed the deploying fabric rest api call and the assert rules for it in all the IT cases in the module inventory/vrf/network.

### CODE REVIEW

File: tests/integration/targets/dcnm_vrf/tests/dcnm/query.yaml [hunk 1]
```diff
@@ -90,6 +90,23 @@
...
+    fabric: "{{ ansible_it_fabric }}"
+    state: query
+  register: result
+
+- assert:
+    that:
+    - 'result.changed == false'
+    - 'result.response[0].fabric == "{{ ansible_it_fabric }}"'
+    - 'result.response[0].vrfName == "ansible-vrf-int1"'
+    - 'result.response[0].vrfStatus == "DEPLOYED"'
```
Comment (mikewiebe) [inline tests/integration/targets/dcnm_vrf/tests/dcnm/query.yaml:108]: Does the `vrfStatus` get returned if you query WITH the `config:` element?  Same question for network.

Comment (mikewiebe) [reply tests/integration/targets/dcnm_vrf/tests/dcnm/query.yaml:108]: It should if it does not.

Comment (kharicha) [reply tests/integration/targets/dcnm_vrf/tests/dcnm/query.yaml:108]: No, the status for both vrf and network comes only when we enquire about "all" the vrf/network present if specified with a specific vrf/network name the status is not populated.

Comment (mikewiebe) [reply tests/integration/targets/dcnm_vrf/tests/dcnm/query.yaml:108]: Why is that?  Is there no way to get the status when a user only wants to know about a specific set of vrfs or networks?  Seems like this is something we should support.

**Pull request closed without merging**